src/ruby/puppetserver-lib/puppet/server/network/http/handler.rb

```diff
@@ -36,7 +36,17 @@ def path(request)
   def body(request)
     body = request["body"]
     if body.kind_of?(InputStream)
-      body.to_io.read()
```

Comment:
We may want to move this to where we turn the Java response into a Ruby response object https://github.com/puppetlabs/puppetserver/blob/6.x/src/ruby/puppetserver-lib/puppet/server/http_client.rb#L169 as I'm not 100% sure if the Http::Handler code wraps all Http responses in Puppet.

Reply:
I *think* they are handled by this?  I thought the bottleneck was here:  https://github.com/puppetlabs/puppetserver/blob/750958954896578ad877d43b8bb61eec66620c70/src/ruby/puppetserver-lib/puppet/server/master.rb#L62-L68  and that includes the above addition.  Is there an alternative path?

Reply:
@justinstoller I looked at your comment again, and I'm a little confused.  I think what you linked deals with the response, and this is dealing with the request?  I certainly could be wrong about how this flow works.

Reply:
Oh, you're right. I was just in the HttpClient and was thinking of responses to Ruby requests like [classification](https://github.com/puppetlabs/classifier/blob/main/puppet/lib/puppet/indirector/node/classifier.rb#L125), but those strings are created in the Java http-client library not through clojure processing in server itself.